tests/Application/Commands/RedistibuteMatchFundsTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MatchBot\Tests\Application\Commands;
+
+use MatchBot\Application\Commands\RedistributeMatchFunds;
+use MatchBot\Application\HttpModels\DonationCreate;
+use MatchBot\Domain\Campaign;
+use MatchBot\Domain\CampaignFunding;
+use MatchBot\Domain\CampaignFundingRepository;
+use MatchBot\Domain\Donation;
+use MatchBot\Domain\DonationRepository;
+use MatchBot\Domain\Pledge;
+use MatchBot\Tests\TestCase;
+use Prophecy\Argument;
+use Prophecy\Prophecy\ObjectProphecy;
+use Psr\Log\NullLogger;
+use Symfony\Component\Console\Tester\CommandTester;
+use Symfony\Component\Lock\LockFactory;
+
+class RedistibuteMatchFundsTest extends TestCase
+{
+    public function testNoEligibleDonations(): void
+    {
+        $donationRepoProphecy = $this->prophesize(DonationRepository::class);
+        $donationRepoProphecy->findWithMatchingWhichCouldBeReplacedWithHigherPriorityAllocation(
+            Argument::type(\DateTimeImmutable::class)
+        )
+            ->willReturn([])
+            ->shouldBeCalledOnce();
+        $donationRepoProphecy->releaseMatchFunds(Argument::type(Donation::class))->shouldNotBeCalled();
+
+        $commandTester = new CommandTester($this->getCommand(
+            $this->prophesize(CampaignFundingRepository::class),
+            $donationRepoProphecy,
+        ));
+        $commandTester->execute([]);
+
+        $expectedOutputLines = [
+            'matchbot:redistribute-match-funds starting!',
+            'Checked 0 donations and redistributed matching for 0',
+            'matchbot:redistribute-match-funds complete!',
+        ];
+        $this->assertEquals(implode("\n", $expectedOutputLines) . "\n", $commandTester->getDisplay());
+        $this->assertEquals(0, $commandTester->getStatusCode());
+    }
+
+    public function testOneDonationHasChampFundsUsedAndCouldBeAssignedPledge(): void
+    {
+        $donationRepoProphecy = $this->prophesize(DonationRepository::class);
+        $donationRepoProphecy->findWithMatchingWhichCouldBeReplacedWithHigherPriorityAllocation(Argument::type(\DateTimeImmutable::class))->willReturn([
+            Donation::fromApiModel(new DonationCreate(
+                currencyCode: 'GBP',
+                donationAmount: '10',
+                projectId: 'project-id',
+                psp: 'stripe',
+            ), $this->getMinimalCampaign()),
+        ]);
+        $donationRepoProphecy->releaseMatchFunds(Argument::type(Donation::class))
+            ->shouldBeCalledOnce();
+        $donationRepoProphecy->allocateMatchFunds(Argument::type(Donation::class))
+            ->shouldBeCalledOnce();
+        $donationRepoProphecy->push(Argument::type(Donation::class), false)
+            ->shouldBeCalledOnce();
```

Comment:
I think we might as well be specific about which donation should be operated on here and in similar tests instead of just relying on the type - Psalm checks types for us, here it'd be good to test specific values. I haven't run this code but I think this should work:

Also gets a bit more concise:

```suggestion

$donation =  Donation::fromApiModel(new DonationCreate(
                currencyCode: 'GBP',
                donationAmount: '10',
                projectId: 'project-id',
                psp: 'stripe',
            );
            $this->getMinimalCampaign())$donationRepoProphecy->findWithMatchingWhichCouldBeReplacedWithHigherPriorityAllocation(Argument::type(\DateTimeImmutable::class))->willReturn([
            $donation,
        ]);
        $donationRepoProphecy->releaseMatchFunds($donation)->shouldBeCalledOnce();
        $donationRepoProphecy->allocateMatchFunds($donation)->shouldBeCalledOnce();
        $donationRepoProphecy->push($donation, false)->shouldBeCalledOnce();
```